src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py

```diff
@@ -0,0 +1,313 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+# noqa: E501
+import asyncio
+import logging
+from typing import Any, Callable, Dict, List
+
+from tqdm import tqdm
+
+from promptflow._sdk._telemetry import ActivityType, monitor_operation
+
+from ._conversation import CallbackConversationBot, ConversationBot, ConversationRole
+from ._conversation._conversation import simulate_conversation
+from ._model_tools import (
+    CONTENT_HARM_TEMPLATES_COLLECTION_KEY,
+    AdversarialTemplateHandler,
+    AsyncHTTPClientWithRetry,
+    ManagedIdentityAPITokenManager,
+    ProxyChatCompletionsModel,
+    RAIClient,
+    TokenScope,
+)
+from ._utils import JsonLineList
+
+logger = logging.getLogger(__name__)
+
+
+class AdversarialSimulator:

```

Comment:
@minthigpen what are other major simulation scenarios we need to support down the road? From high level I'm thinking we should create one class per simulation scenario, hence the class names become critical, because we don't want this class name to step into other scenarios for example. It may be nice to have a tree diagram for all the scenarios we want to support.

Reply:
No additional scenarios will need to be supported in the near future. The ones we support for adversarial simulator (for text) will be it for now... but we do need to think about the expansion to image generation + multi-modal soon for adversarial simulator.